protobuf/lib/src/protobuf/coded_buffer.dart

```diff
@@ -129,33 +129,65 @@ void _mergeFromCodedBufferReader(BuilderInfo meta, _FieldSet fs,
       case PbFieldType._REPEATED_BOOL:
         final list = fs._ensureRepeatedField(meta, fi);
         if (wireType == WIRETYPE_LENGTH_DELIMITED) {
-          _readPacked(input, () => list.add(input.readBool()));
+          final limit = input.readInt32();
+          if (limit != 0) {
+            list._checkModifiable('add');
+            // No need to check the element as for `bool` fields we only need to
+            // check that the value is not null, and we know in `add` below that
+            // the value isn't null (`readBool` doesn't return `null`).
+            input._withLimit(limit, () {
+              while (!input.isAtEnd()) {
+                list._addUnchecked(input.readBool());
+              }
+            });
+          }
         } else {
-          list.add(input.readBool());
+          list._checkModifiable('add');
+          list._addUnchecked(input.readBool());
         }
         break;
       case PbFieldType._REPEATED_BYTES:
         final list = fs._ensureRepeatedField(meta, fi);
-        list.add(input.readBytes());
+        list._checkModifiable('add');
```

Comment:
It seems to be hand-inlined code, omitting the `_check` call. It seems not very nice we have to copy&paste this many times.

Why not pass in a `bool` whether to check or not, and force inline the function. E.g.

```dart
_addInternal(input.readBytes(), omitCheck: true);

@pragma('vm/dart2js/dart2wasm:prefer-inline')
_addInternal(E value, {bool omitCheck = false}) {
  _checkModifiable(value, 'add');
  if (!omitCheck) _check(value);
  _wrappedList.add(value);
}
```

Reply:
This function wouldn't work below where we check for modifiable once before adding the elements, and then never check it while adding the elements.

We could have this version:

```dart
_addInternal(E value, {bool omitElementCheck = false, bool omitModifiableCheck = false}) {
  if (!omitModifiableCheck) _checkModifiable(value, 'add');
  if (!omitElementCheck) _check(value);
  _wrappedList.add(value);
}
```

But we would still need a `_checkModifiable` call in packed fields as we want to run it once.

Overall not sure if this version is better than the current..